data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -94,6 +96,13 @@ public KeyValueProcessor(final PluginMetrics pluginMetrics, final KeyValueProces
         if(keyValueProcessorConfig.getIncludeKeys() != null) {
             includeKeysSet.addAll(keyValueProcessorConfig.getIncludeKeys());
         }
+
+        if(keyValueProcessorConfig.getTransformKey() != null
+                && !keyValueProcessorConfig.getTransformKey().isEmpty()) {
+            if(!validTransformOptionSet.contains(keyValueProcessorConfig.getTransformKey())) {
+                throw new IllegalArgumentException("transform_key is not a valid option");
```

Comment:
The message here might be confusing. 
```suggestion
                throw new IllegalArgumentException(String.format("The transform_key value: %s is not a valid option", keyValueProcessorConfig.getTransformKey()));
```